### PR TITLE
Default fleet policy removal

### PR DESCRIPTION
**Pull request by lykkin**
## What is the problem this PR solves?
Remove references to the default field on the policy; use a default policy id instead.

## Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

Agent changes: https://github.com/elastic/beats/pull/30256

Related discussion: 
Closes https://github.com/elastic/beats/issues/29774

### CODE REVIEW

**Review by AndersonQ — COMMENTED** [review]
LGTM, just fix the lining issues. However I cannot access the impact of this change.

**Comment by lykkin** [comment]
Looks like some of the linting errors are in generated files, so I'm going to track those back to the generator to allow renaming rules of struct fields.

**Comment by lykkin** [comment]
finally, it seems the linter is satisfied

**Comment by ph** [comment]
Looking at this now, thanks for the changer in https://github.com/elastic/go-json-schema-generate/pull/1

**Review by ph — COMMENTED** [review]
Oh linters, so evil :(

@lykkin I am curious where these linters errors were raised in the github action or locally?
Looking at your commits, I believe this one is the required changes https://github.com/elastic/fleet-server/pull/1157/commits/b06c48b2af752b49e9964bd603ac7407d35145b2  I would have expected the github action to really complain on a subset.

**Comment by ph** [comment]
@lykkin The code here looks fine to me, I haven't approved yet because I want to make sure the other is approved and I can validate it.

**Comment by lykkin** [comment]
> Oh linters, so evil :(
> 
> @lykkin I am curious where these linters errors were raised in the github action or locally? Looking at your commits, I believe this one is the required changes [b06c48b](https://github.com/elastic/fleet-server/commit/b06c48b2af752b49e9964bd603ac7407d35145b2) I would have expected the github action to really complain on a subset.

The issue was in the generated code needing to be updated to the linter standards. the `Id -> ID` and `Api -> API` change on the model required a change in the generated names of all the fields, which propagated to other files. There is still one field (namely `model.Agent.Id` I believe) that didn't get hit by the transformation on the schema generation, but I think this is a good start.

File: internal/pkg/policy/self.go [hunk 1]
```diff
@@ -201,31 +203,32 @@ func (m *selfMonitorT) updateStatus(ctx context.Context) (proto.StateObserved_St
...
-		if m.policyId == "" {
-			m.reporter.Status(proto.StateObserved_STARTING, "Waiting on default policy with Fleet Server integration", nil)
+		if m.policyID == "" {
+			err = m.reporter.Status(proto.StateObserved_STARTING, "Waiting on default policy with Fleet Server integration", nil)
 		} else {
-			m.reporter.Status(proto.StateObserved_STARTING, fmt.Sprintf("Waiting on policy with Fleet Server integration: %s", m.policyId), nil)
+			err = m.reporter.Status(proto.StateObserved_STARTING, fmt.Sprintf("Waiting on policy with Fleet Server integration: %s", m.policyID), nil)
 		}
-		return proto.StateObserved_STARTING, nil
+		return proto.StateObserved_STARTING, err
```
Comment (michel-laterman) [inline internal/pkg/policy/self.go:215]: Should we be concerned about it returning a non-failed status with an error?

Comment (lykkin) [reply internal/pkg/policy/self.go:215]: it looks like the status update code doesn't return meaningful errors (i.e. always `nil`), but I like the idea for posterity.

**Review by michel-laterman — APPROVED** [review]

**Review by ph — COMMENTED** [review]
LGTM

**Pull request merged**